### PR TITLE
Modifications

### PR DESCRIPTION
**Pull request by billsacks**
Various suggestions and fixes. We can talk tomorrow if you have any questions.

In addition, I noticed a lot of errors, mostly in the biogeochem directory. Rather than fixing all of them, here is how I found them, so that you can reproduce the issue and confirm that it's fixed:

(1) Errors in file names: I created a file that started off as a join of the `dir_name` and `old_fname` columns, then I modified it so that each line looked like this: `ls src/main/column_varcon.F90 > /dev/null`, etc. I then ran this file. This printed a list of files not found:

```
ls: src/biogeochem/CNSharedParams.F90: No such file or directory
ls: src/biogeochem/EDBGCDyn.F90: No such file or directory
ls: src/biogeochem/MEGANFactors.F90: No such file or directory
ls: src/biogeochem/DynConsBiogeochem.F90: No such file or directory
ls: src/biogeochem/CNFireLi2014.F90: No such file or directory
ls: src/biogeochem/CNRootDyn.F90: No such file or directory
ls: src/biogeochem/CNFireMethod.F90: No such file or directory
ls: src/biogeochem/CNFireEmissions.F90: No such file or directory
ls: src/biogeochem/CNPrecisionControl.F90: No such file or directory
ls: src/biogeochem/CNCStateUpdate2.F90: No such file or directory
ls: src/biogeochem/Ch4.F90: No such file or directory
ls: src/biogeochem/CNFireFactory.F90: No such file or directory
ls: src/biogeochem/FireEmisFactors.F90: No such file or directory
ls: src/biogeochem/CNVegStructUpdate.F90: No such file or directory
ls: src/biogeochem/DynCNDV.F90: No such file or directory
ls: src/biogeochem/CNAnnualUpdate.F90: No such file or directory
ls: src/biogeochem/CNCStateUpdate3.F90: No such file or directory
ls: src/biogeochem/NutrientCompetitionFactory.F90: No such file or directory
ls: src/biogeochem/CNVegComputeSeed.F90: No such file or directory
ls: src/biogeochem/CNFireNoFire.F90: No such file or directory
ls: src/biogeochem/CNBalanceCheck.F90: No such file or directory
ls: src/biogeochem/CNMResp.F90: No such file or directory
ls: src/biogeochem/CNNStateUpdate3.F90: No such file or directory
ls: src/biogeochem/CNFireLi2016.F90: No such file or directory
ls: src/biogeochem/CNSpecies.F90: No such file or directory
ls: src/biogeochem/CNGResp.F90: No such file or directory
ls: src/biogeochem/CNFireBase.F90: No such file or directory
ls: src/biogeochem/CNDVLight.F90: No such file or directory
ls: src/biogeochem/CNNStateUpdate2.F90: No such file or directory
ls: src/biogeochem/VOCEmission.F90: No such file or directory
ls: src/biogeochem/NutrientCompetitionMethod.F90: No such file or directory
ls: src/biogeochem/CNDriver.F90: No such file or directory
ls: src/biogeochem/CNNDynamics.F90: No such file or directory
ls: src/biogeochem/CNPhenology.F90: No such file or directory
ls: src/biogeochem/CNC14Decay.F90: No such file or directory
ls: src/biogeochem/SatellitePhenology.F90: No such file or directory
ls: src/biogeochem/DynHarvest.F90: No such file or directory
ls: src/biogeochem/CNProducts.F90: No such file or directory
ls: src/biogeochem/NutrientCompetitionCLM45default.F90: No such file or directory
ls: src/biogeochem/CNDVEstablishment.F90: No such file or directory
ls: src/biogeochem/CNCStateUpdate1.F90: No such file or directory
ls: src/biogeochem/CNFUN.F90: No such file or directory
ls: src/biogeochem/NutrientCompetitionFlexibleCN.F90: No such file or directory
ls: src/biogeochem/CNNStateUpdate1.F90: No such file or directory
ls: src/biogeochem/DUST.F90: No such file or directory
ls: src/biogeochem/CNDVDriver.F90: No such file or directory
ls: src/biogeochem/CNGapMortality.F90: No such file or directory
```

(2) Errors in module names: I created a file with just the `old_string` column, then ran the following python code from the src subdirectory:

```python
#!/usr/local/bin/python3

import subprocess
import os

with open('/Users/sacks/temporary/modnames.csv', 'r') as f:
    for string in f:
        string = string.strip()
        grep_string = "^module *{} *$".format(string)
        result = subprocess.run(["git", "grep", "-i", grep_string], capture_output=True)
        if result.returncode:
            print("<{}> not found".format(grep_string))
```

This again resulted in a long list of files. Many are the same as the files not found, but not all (e.g., file initInterp.F90 has a module named initInterpMod).

### CODE REVIEW

File: ctsm_filenames.csv [hunk 1]
```diff
@@ -1,12 +1,12 @@
 ﻿,dir_name,old_fname,new_fname,old_string,new_string
 0,/glade/scratch/negins/ctsm_0110/src/main,column_varcon.F90,ctsm_ColumnVarCon.F90,column_varcon,ctsm_ColumnVarCon
 1,/glade/scratch/negins/ctsm_0110/src/main,initGridCellsMod.F90,ctsm_InitGridCells.F90,initGridCellsMod,ctsm_InitGridCells
-2,/glade/scratch/negins/ctsm_0110/src/main,clm_varsur.F90,ctsm_VarSur.F90,clm_varsur,ctsm_VarSur
+2,/glade/scratch/negins/ctsm_0110/src/main,clm_varsur.F90,ctsm_VarSur.F90,clm_instur,ctsm_VarSur
```
Comment (billsacks) [inline ctsm_filenames.csv:4]: Not sure why this module name disagreed with the file name, but it did.

File: ctsm_filenames.csv [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 0,/glade/scratch/negins/ctsm_0110/src/main,column_varcon.F90,ctsm_ColumnVarCon.F90,column_varcon,ctsm_ColumnVarCon
 1,/glade/scratch/negins/ctsm_0110/src/main,initGridCellsMod.F90,ctsm_InitGridCells.F90,initGridCellsMod,ctsm_InitGridCells
-2,/glade/scratch/negins/ctsm_0110/src/main,clm_varsur.F90,ctsm_VarSur.F90,clm_varsur,ctsm_VarSur
+2,/glade/scratch/negins/ctsm_0110/src/main,clm_varsur.F90,ctsm_VarSur.F90,clm_instur,ctsm_VarSur
 3,/glade/scratch/negins/ctsm_0110/src/main,clm_varpar.F90,ctsm_VarPar.F90,clm_varpar,ctsm_VarPar
 4,/glade/scratch/negins/ctsm_0110/src/main,filterColMod.F90,ctsm_FilterCol.F90,filterColMod,ctsm_FilterCol
 5,/glade/scratch/negins/ctsm_0110/src/main,GetGlobalValuesMod.F90,ctsm_GetGlobalValues.F90,GetGlobalValuesMod,ctsm_GetGlobalValues
 6,/glade/scratch/negins/ctsm_0110/src/main,restFileMod.F90,ctsm_RestFile.F90,restFileMod,ctsm_RestFile
-7,/glade/scratch/negins/ctsm_0110/src/main,accumulMod.F90,ctsm_Accumul.F90,accumulMod,ctsm_Accumul
+7,/glade/scratch/negins/ctsm_0110/src/main,accumulMod.F90,ctsm_Accumulators.F90,accumulMod,ctsm_Accumulators
```
Comment (billsacks) [inline ctsm_filenames.csv:9]: Here and in a few other places, I felt like a slightly longer name seemed more appropriate with the dropping of 'Mod'. In some cases, a plural form felt more appropriate.

File: ctsm_filenames.csv [hunk 3]
```diff
@@ -40,29 +40,29 @@
...
-62,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/utils,spmdMod_stub.F90,ctsm_SpmdStub.F90,spmdMod_stub,ctsm_SpmdStub
-63,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/dyn_subgrid,dynFileMod_stub.F90,ctsm_DynFileStub.F90,dynFileMod_stub,ctsm_SpmdStub
+58,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/csm_share,seq_comm_mct.F90,seq_comm_mct.F90,seq_comm_mct,seq_comm_mct
+59,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/csm_share,shr_mpi_mod_stub.F90,shr_mpi_mod_stub.F90,shr_mpi_mod,shr_mpi_mod
+60,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/csm_share,mct_mod_stub.F90,mct_mod_stub.F90,mct_mod,mct_mod
+61,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/utils,clmfates_paraminterfaceMod_stub.F90,ctsm_FatesParamInterface_stub.F90,CLMFatesParamInterfaceMod,ctsm_FatesParamInterface
+62,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/utils,spmdMod_stub.F90,ctsm_Spmd_stub.F90,spmdMod,ctsm_Spmd
+63,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/dyn_subgrid,dynFileMod_stub.F90,ctsm_DynFile_stub.F90,dynFileMod,ctsm_DynFile
```
Comment (billsacks) [inline ctsm_filenames.csv:65]: This stub stuff deserves some explanation of my thinking. A key thing to understand here is that these stub modules are compiled in place of the real ones via build-time logic in the unit test build. So it is critical that the module name be the same as the real module name: we can't have "stub" on the module name. As for the file names: I don't feel strongly about the changes I suggested here, but I have a general rule that, when there is a part of a name that has a different characteristic from the rest of the name, then the separator/convention for that part should differ. That's pretty vague. What I mean here is, for example: We have a file named `ctsm_GetGlobalValues.F90` and then another file that is a stub of that one. In the stub's file name, I wanted the base name (`ctsm_GetGlobalValues`) to be grouped together, and the fact that it's a stub to be separated from that base name. I felt that `ctsm_GetGlobalValues_stub` accomplished this goal better than `ctsm_GetGlobalValuesStub`. But again, I don't feel strongly about this.

Comment (billsacks) [reply ctsm_filenames.csv:65]: Also, some of these are stubs of share code, which doesn't get the `ctsm_` prefix.

File: ctsm_filenames.csv [hunk 4]
```diff
@@ -73,10 +73,10 @@
 71,/glade/scratch/negins/ctsm_0110/src/utils,clm_nlUtilsMod.F90,ctsm_NlUtils.F90,clm_nlUtilsMod,ctsm_NlUtils
 72,/glade/scratch/negins/ctsm_0110/src/utils,AnnualFluxDribbler.F90,ctsm_AnnualFluxDribbler.F90,AnnualFluxDribbler,ctsm_AnnualFluxDribbler
 73,/glade/scratch/negins/ctsm_0110/src/utils,domainMod.F90,ctsm_Domain.F90,domainMod,ctsm_Domain
-74,/glade/scratch/negins/ctsm_0110/src/utils,clmfates_paraminterfaceMod.F90,ctsm_FatesParamInterface.F90,clmfates_paraminterfaceMod,ctsm_FatesParamInterface
+74,/glade/scratch/negins/ctsm_0110/src/utils,clmfates_paraminterfaceMod.F90,ctsm_FatesParamInterface.F90,CLMFatesParamInterfaceMod,ctsm_FatesParamInterface
```
Comment (billsacks) [inline ctsm_filenames.csv:76]: This FATES module and a few other modules had module names that differed from the file name. I have corrected some of these, but not all (see my overall comment on the PR).

File: ctsm_filenames.csv [hunk 5]
```diff
@@ -166,18 +166,18 @@
 164,/glade/scratch/negins/ctsm_0110/src/biogeochem,CNDVDriver.F90,ctsm_CNDVDriver.F90,CNDVDriver,ctsm_CNDVDriver
 165,/glade/scratch/negins/ctsm_0110/src/biogeochem,CNGapMortality.F90,ctsm_CNGapMortality.F90,CNGapMortality,ctsm_CNGapMortality
 166,/glade/scratch/negins/ctsm_0110/src/biogeochem,CropType.F90,ctsm_CropType.F90,CropType,ctsm_CropType
-167,/glade/scratch/negins/ctsm_0110/src/init_interp,initInterpMultilevelBase.F90,ctsm_InitInterpMultiLevelBase.F90,initInterpMultilevelBase,ctsm_InitInterpMultiLevelBase
+167,/glade/scratch/negins/ctsm_0110/src/init_interp,initInterpMultilevelBase.F90,ctsm_InitInterpMultilevelBase.F90,initInterpMultilevelBase,ctsm_InitInterpMultilevelBase
```
Comment (billsacks) [inline ctsm_filenames.csv:169]: Here and elsewhere: Multilevel is one word and I felt it reads better as `Multilevel` than `MultiLevel`

**Comment by billsacks** [comment]
@negin513 this is ready for review and merge (if you agree with my changes).

I'm also going to make some comments back in the main issue about some things I thought of that aren't directly related to this spreadsheet.

File: ctsm_filenames.csv (hunk 2 above)
Comment (negin513) [reply ctsm_filenames.csv:9]: Perfect! I agree with the longer names.

File: ctsm_filenames.csv (hunk 5 above)
Comment (negin513) [reply ctsm_filenames.csv:169]: Thanks!

File: ctsm_filenames.csv (hunk 1 above)
Comment (negin513) [reply ctsm_filenames.csv:4]: What do you think about making the file names and module names consistent?

File: ctsm_filenames.csv [hunk 6]
```diff
@@ -46,23 +46,23 @@
...
 51,/glade/scratch/negins/ctsm_0110/src/cpl/mct,lnd_comp_mct.F90,lnd_comp_mct.F90,lnd_comp_mct,lnd_comp_mct
 52,/glade/scratch/negins/ctsm_0110/src/cpl/nuopc,lnd_import_export.F90,lnd_import_export.F90,lnd_import_export,lnd_import_export
 53,/glade/scratch/negins/ctsm_0110/src/cpl/nuopc,lnd_shr_methods.F90,lnd_shr_methods.F90,lnd_shr_methods,lnd_shr_methods
 54,/glade/scratch/negins/ctsm_0110/src/cpl/nuopc,lnd_comp_nuopc.F90,lnd_comp_nuopc.F90,lnd_comp_nuopc,lnd_comp_nuopc
-55,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/main,GetGlobalValuesMod_stub.F90,ctsm_GetGlobalValues_stub.F90,GetGlobalValuesMod_stub,ctsm_GetGlobalValues_stub
-56,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/main,histFileMod_stub.F90,ctsm_HistFile_stub.F90,histFileMod_stub,ctsm_HistFile_stub
+55,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/main,GetGlobalValuesMod_stub.F90,ctsm_GetGlobalValues_stub.F90,GetGlobalValuesMod,ctsm_GetGlobalValues
+56,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/main,histFileMod_stub.F90,ctsm_HistFile_stub.F90,histFileMod,ctsm_HistFile
```
Comment (negin513) [inline ctsm_filenames.csv:58]: Again, here there is the inconsistency between filenames and module names. I think using consistent names between modules and files might be beneficial.

File: ctsm_filenames.csv (hunk 4 above)
Comment (negin513) [reply ctsm_filenames.csv:76]: Thanks for pointing this out! I will make sure to find the modules with different names from filenames. But with the new module names, there will be consistency between file names and module names.

File: ctsm_filenames.csv (hunk 1 above)
Comment (billsacks) [reply ctsm_filenames.csv:4]: Yes, we should make them consistent moving forward. My point in pointing this out is that your search & replaces won't work unless you have the correct old names.

Comment (negin513) [reply ctsm_filenames.csv:4]: Thanks for pointing this out but the new name will be similar to filenames: 
i.e. module name will be changed from `clm_instur` to ctsm_VarSur`
Is my understanding correct?

Comment (billsacks) [reply ctsm_filenames.csv:4]: Yes

File: ctsm_filenames.csv (hunk 5 above)
Comment (negin513) [reply ctsm_filenames.csv:169]: I am slightly more inclined to use `MultiLevel` for better readability. The word multi-level itself uses the dash in between for easier reading. 
For example, I think  `ctsm_InitInterpMultiLevelBase.F90` is easier to read than `ctsm_InitInterpMultilevelBase.F90`.
But I don't feel very strongly about this.

Comment (billsacks) [reply ctsm_filenames.csv:169]: Multilevel is a dictionary word without the hyphen (https://www.merriam-webster.com/dictionary/multilevel), but I really am fine either way on this, so feel free to go with your preference.

**Comment by negin513** [comment]
> (1) Errors in file names: I created a file that started off as a join of the `dir_name` and `old_fname` columns, then I modified it so that each line looked like this: `ls src/main/column_varcon.F90 > /dev/null`, etc. I then ran this file. This printed a list of files not found:

Thanks for pointing this out Bill. I ran the script to rename the files once under `src/biogeochem/` but forgot to change them back to the original version. I will correct all of these. 
Most of them (if not all) are missing a mod which was removed because of the renaming scripts. I remembered to remove the `ctsm_` prefix from the filenames in this directory but forgot about other changes. 

> (2) Errors in module names: I created a file with just the `old_string` column, then ran the following python code from the src subdirectory:
> 
> ```python
> #!/usr/local/bin/python3
> 
> import subprocess
> import os
> 
> with open('/Users/sacks/temporary/modnames.csv', 'r') as f:
>     for string in f:
>         string = string.strip()
>         grep_string = "^module *{} *$".format(string)
>         result = subprocess.run(["git", "grep", "-i", grep_string], capture_output=True)
>         if result.returncode:
>             print("<{}> not found".format(grep_string))
> ```
> 
> This again resulted in a long list of files. Many are the same as the files not found, but not all (e.g., file initInterp.F90 has a module named initInterpMod).

Similarly part of this inconsistency is caused by my mistake in running the renaming scripts on `src/biogeochem`. However, there are some modules that are named different from the file names (you pointed out many of them). I will make sure to have all the module names correctly in this table.

**Pull request merged**